lib/pages/portal/view/website_view.dart

```diff
@@ -323,33 +403,45 @@ class _WebsiteViewState extends State<WebsiteView> {
 }
 
 class WebsiteIcon extends StatelessWidget {
-  const WebsiteIcon({this.website, this.canEdit, this.size, this.onTap});
+  const WebsiteIcon(
+      {this.website, this.canEdit, this.size, this.image, this.onTap});
 
   final Website website;
   final bool canEdit;
   final double size;
+
+  // image represents the new website icon in case the user uploaded it
```

Comment:
That's not what this does, that's just how you're currently using it. Documentation should describe what the behaviour of the thing it's describing is, not where or how it's used, as usages can change in time without the implementation (and corresponding comment) changing.

Try to think of `WebsiteIcon` as a widget of its own, not at its current usages. It's simply a widget that displays a website circle with the label below, it knows nothing about user uploads or anything of the sort. Specifically, what this variable does is it just overrides the default image icon. Basically by default we fetch the icon from storage using `website.iconPath`, but if `image` is provided, we ignore that and use `image` instead. Please use the comment I suggested in my initial review (below).